mono/unit-tests/test-mono-string.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * test-mono-string.c: Unit test for runtime MonoString* manipulation.
+ */
+
+#include "config.h"
+#include <stdio.h>
+#include "metadata/object-internals.h"
+#include "mini/jit.h"
+
+static int
+new_string_ok (void)
+{
+	MonoError error;
+	MonoString *s = mono_string_new_checked (mono_domain_get (), "abcd", &error);
+	mono_error_assert_ok (&error);
+	gunichar2* c = mono_string_chars (s);
+
+	g_assert ((c[0] == (gunichar2)'a') &&
+		  (c[1] == (gunichar2)'b') &&
+		  (c[2] == (gunichar2)'c') &&
+		  (c[3] == (gunichar2)'d') &&
+		  (c[4] == (gunichar2)0));
+	return 0;
+}
+
+static int
+new_string_utf8 (void)
+{
+	MonoError error;
+	const gunichar2 snowman = 0x2603;
+	const char bytes[] = { 0xE2, 0x98, 0x83, 0x00 }; /* U+2603 NUL */
```

Comment:
Can also be "\xE2\x98\x83" but same thing. Also should be const static to demonstrate good form, even if it is only test code. This non-static local actually gets room on the stack and copied to it.